jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -1151,6 +1156,127 @@ ecma_builtin_object_object_is (ecma_value_t arg1, /**< routine's first argument
   return ecma_op_same_value (arg1, arg2) ? ECMA_VALUE_TRUE : ECMA_VALUE_FALSE;
 } /* ecma_builtin_object_object_is */
 
+/**
+ * The Object object's 'fromEntries' routine
+ *
+ * See also:
+ *          ECMA-262 v10, 19.1.2.7
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_object_from_entries (ecma_value_t iterator) /**< object's iterator */
+{
+  /* 1. */
+  if (ECMA_IS_VALUE_ERROR (ecma_op_check_object_coercible (iterator)))
+  {
+    return ECMA_VALUE_ERROR;
+  }
+  /* 2 */
+  ecma_object_t *object_prototype_p = ecma_builtin_get (ECMA_BUILTIN_ID_OBJECT_PROTOTYPE);
+  ecma_object_t *obj_p = ecma_create_object (object_prototype_p, 0, ECMA_OBJECT_TYPE_GENERAL);
+
+  /* 6 */
+  ecma_value_t next_method;
+  ecma_value_t result = ecma_op_get_iterator (iterator, ECMA_VALUE_SYNC_ITERATOR, &next_method);
+
+  if (ECMA_IS_VALUE_ERROR (result))
+  {
+    ecma_free_value (next_method);
```

Comment:
This should be free, since `next_method` is initialized only if the result of `GetIterator` operation is not an abrupt completion. Please add a testcase as well.

Reply:
Sorry let me correct myself. This should **NOT** be freed,....